src/plugins/discover/public/application/main/hooks/use_adhoc_data_views.ts

```diff
@@ -37,11 +40,19 @@ export const useAdHocDataViews = ({
   dataViews: DataViewsContract;
   filterManager: FilterManager;
   toastNotifications: ToastsStart;
+  trackUiMetric?: (metricType: string, eventName: string | string[], count?: number) => void;
 }) => {
   const [adHocDataViewList, setAdHocDataViewList] = useState<DataView[]>(
     !dataView.isPersisted() ? [dataView] : []
   );
 
+  useEffect(() => {
+    if (trackUiMetric && adHocDataViewList.length) {
+      trackUiMetric(METRIC_TYPE.CLICK, ADHOC_DATA_VIEW_CLICK_EVENT, adHocDataViewList.length);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [adHocDataViewList]);
```

Comment:
I guess we should send an event only if current data view on Discover is ad-hoc, not for the available list of them. On Lens page it's a bit different: several data views can be active and referenced in vis layers.

Reply:
Changed.